cg/services/illumina_services/illumina_post_processing_service/illumina_post_processing_service.py

```diff
@@ -115,10 +116,32 @@ def store_sequencing_data_in_status_db(
         sequencing_run: IlluminaSequencingRun = self.store_illumina_sequencing_run(
             run_directory_data=run_directory_data, flow_cell=flow_cell
         )
-        self.store_illumina_sample_sequencing_metrics(
-            run_directory_data=run_directory_data, sequencing_run=sequencing_run
+        sample_metrics: list[IlluminaSampleSequencingMetricsDTO] = (
+            self.store_illumina_sample_sequencing_metrics(
+                run_directory_data=run_directory_data, sequencing_run=sequencing_run
+            )
         )
         self.status_db.commit_to_store()
+        self.update_samples_for_metrics(
+            sample_metrics=sample_metrics, sequencing_run=sequencing_run
+        )
+
+    def update_samples_for_metrics(
+        self,
+        sample_metrics: list[IlluminaSampleSequencingMetricsDTO],
+        sequencing_run: IlluminaSequencingRun,
+    ):
+        unique_samples_on_run: list[str] = self.get_unique_samples_from_run(sample_metrics)
+        for sample_id in unique_samples_on_run:
+            self.status_db.update_sample_reads_illumina(internal_id=sample_id)
+            self.status_db.update_sample_sequenced_at(
+                sample_id, date=sequencing_run.sequencing_completed_at
+            )
+
+    @staticmethod
+    def get_unique_samples_from_run(sample_metrics: list[IlluminaSampleSequencingMetricsDTO]):
+        """Get unique samples from the run."""
+        return list({sample_metric.sample_id for sample_metric in sample_metrics})
```

Comment:
Return type hint and, why not just return a set?
```suggestion
    def get_unique_samples_from_run(sample_metrics: list[IlluminaSampleSequencingMetricsDTO])-> set(str):
        """Get unique samples from the run."""
        return {sample_metric.sample_id for sample_metric in sample_metrics}
```

Reply:
Fixed